src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Compiler/CachePoolClearerPassTest.php

```diff
@@ -55,7 +55,12 @@ public function testPoolRefsAreWeak()
         $container->setAlias('clearer_alias', 'clearer');
 
         $pass = new RemoveUnusedDefinitionsPass();
-        $pass->setRepeatedPass(new RepeatedPass(array($pass)));
+        foreach ($container->getCompiler()->getPassConfig()->getRemovingPasses() as $removingPass) {
+            if ($removingPass instanceof RepeatedPass) {
+                $pass->setRepeatedPass(new RepeatedPass(array($pass)));
```

Comment:
shouldn't you use `$removingPass` instead of `$pass` in the `array` ? Otherwise it seems that `$removingPass` is only used to detect if it's a `RepeatedPass` ?

(disclaimer : I don't really know what the `RepeatedPass` does, that's just out of curiosity, as it looks strange)

Reply:
ah nevermind didn't see the `break` below, so it does the same thing as before, except it adds the repeated pass only if there's a `RepeatedPass` pass in the passes (... right ?).

Reply:
> it adds the repeated pass only if there's a RepeatedPass pass in the passes

correct, that's part of the BC layer